src/python/pants/backend/explorer/graphql/query/rules.py

```diff
@@ -53,12 +53,12 @@ def filter(query: RulesQuery | None, rules: Iterable[RuleInfo]) -> Iterator[Rule
         name_pattern = query.name_re and re.compile(query.name_re)
         count = 0
         for info in rules:
-            if name_pattern and not re.match(name_pattern, info.name):
+            if name_pattern and not re.search(name_pattern, info.name):
                 continue
+            if query.limit is not None and count >= query.limit:
+                return
```

Comment:
This should run before the `re.search` probably?

Reply:
No, the limit tells how many results we want (at most), if we count excluded entries too, we may end up with no results, even if there could've been some if the limit was high enough.